src/Game/AccountManagement.java

```diff
@@ -114,6 +123,222 @@ public static void register(String user, String pass) {
+	          	 * incl. up to RSC175 was changed to just send the encrypted answers, probably due to collisions
+	          	 * in RSC127 hasher function */
+	        	String answer = Panel.getControlText(Client.panelRecovery, Client.controlRecoveryInput[i]);
+	        	answer = Util.formatString(answer, 50);
+	        	StreamUtil.putByteTo(buffer, (byte)answer.length());
+	            enc_cred_put(buffer, answer, Client.session_id);
+	        }
+
+			StreamUtil.flushPacket();
+			
+			StreamUtil.readStream(); // Unknown what data this contained, client doesn't use it.
+			int response = StreamUtil.readStream();
+			Logger.Game("Recover response: " + response);
+			if (response == 0) {
+				Client.login_screen = Client.SCREEN_USERNAME_PASSWORD_LOGIN;
+				Client.setLoginMessage("", "Sorry, recovery failed. You may try again in 1 hour");
+				Client.failedRecovery = true;
+				return;
+			}
+
+			if (response == 1) {
+				Client.login_screen = Client.SCREEN_USERNAME_PASSWORD_LOGIN;
+				Client.setLoginMessage("", "Your pass has been reset. You may now use the new pass to login");
+				return;
+			}
+
+			Client.login_screen = Client.SCREEN_USERNAME_PASSWORD_LOGIN;
+			Client.setLoginMessage("", "Recovery failed! Attempts exceeded?");
+			return;
+		} catch (Exception e) {
+			e.printStackTrace();
+			Client.setLoginMessage("Check your internet settings", "Sorry! Unable to connect to server.");
+		}
+  }
+  
+  public static void sendPassChange(String oldPwd, String newPwd) {
+	  String oldPass = Client.formatText(oldPwd, 20); 
+	  String newPass = Client.formatText(newPwd, 20);
+	  
+	  StreamUtil.newPacket(25);
+	  
+	  Object buffer = StreamUtil.getStreamBuffer();
+	  enc_cred_put(buffer, oldPass + newPass, Client.session_id);
+	  
+	  StreamUtil.sendPacket();
+  }
+  
+  public static void sendRecoveryQuestions() {
+	  StreamUtil.newPacket(208);
+      Object buffer = StreamUtil.getStreamBuffer();
+
+		for (int idx = 0; idx < 5; ++idx) {
+			String question = Client.controlRecoveryText[idx];
+			if (question == null || question.length() == 0) {
+              question = String.valueOf(idx + 1);
+          }
+          if (question.length() > 50) {
+              question = question.substring(0, 50);
+          }
+          String answer = Panel.getControlText(Client.panelRecoveryQuestions, Client.controlAnswerInput[idx]);
+          answer = Util.formatString(answer, 50);
+          
+          StreamUtil.putByteTo(buffer, (byte)question.length());
+          StreamUtil.putRegStrTo(buffer, question);
+          /** In RSC127 it would have sent an encrypted "hash" like of answers, but seems in later revisions
+      	 * incl. up to RSC175 was changed to just send the encrypted answers, probably due to collisions
+      	 * in RSC127 hasher function */
+          StreamUtil.putByteTo(buffer, (byte)answer.length());
+          enc_cred_put(buffer, answer, Client.session_id);
+		}
+
+		StreamUtil.sendPacket();

```

Comment:
Maybe missing the behaviour of resetting all security answer & question fields here

Reply:
nvm